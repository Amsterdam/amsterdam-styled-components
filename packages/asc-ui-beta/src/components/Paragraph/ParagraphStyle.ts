import { themeColor, breakpoint } from '@amsterdam/asc-ui'
import styled, { css } from 'styled-components'

import { fluidTypoStyle, typographyStyle } from '../../utils/betaUtils'

export interface Props {
  intro?: boolean
}

const ParagraphStyle = styled.p<Props>`
  color: ${themeColor('tint', 'level6')};
  letter-spacing: 0.0125rem;
  font-weight: 500;

  @media screen and ${breakpoint('max-width', 'laptop')} {
    letter-spacing: 0;
  }

  ${({ intro }) =>
    intro &&
    css`
      letter-spacing: 0.025rem;

      @media screen and ${breakpoint('max-width', 'laptop')} {
        font-weight: 900;
        letter-spacing: 0.0125rem;
      }
    `}

  ${fluidTypoStyle}
  ${typographyStyle}
`

export default ParagraphStyle
