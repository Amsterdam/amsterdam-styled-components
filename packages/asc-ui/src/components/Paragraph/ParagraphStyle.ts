import styled, { css } from '@datapunt/asc-core'
import TypographyStyle from '../Typography/TypographyStyle'
import { breakpoint, color } from '../../utils'

export type Props = {
  hasLongText?: boolean
  strong?: boolean
}

export const ParagraphStyleCSS = css<Props>`
  margin-top: 0;
  color: inherit;
  ${({ strong }) =>
    strong &&
    css`
      font-weight: 700;
      color: ${color('tint', 'level7')};
    `}
  ${({ hasLongText }) =>
    hasLongText &&
    css`
      @media screen and ${breakpoint('min-width', 'tabletS')} {
        font-size: 18px;
        line-height: 25px;
      }
    `}
`

export default styled(TypographyStyle)<Props>`
  ${ParagraphStyleCSS}
`
