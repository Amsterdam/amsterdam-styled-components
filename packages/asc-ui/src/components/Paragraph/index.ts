export { default as ParagraphStyle } from './ParagraphStyle'
export { default } from './Paragraph'
