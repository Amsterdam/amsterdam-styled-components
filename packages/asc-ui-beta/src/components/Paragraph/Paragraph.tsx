import type { ElementType, HTMLAttributes } from 'react'
import ParagraphStyle from './ParagraphStyle'

export type Props = {
  as?: ElementType
  strong?: boolean
} & HTMLAttributes<HTMLParagraphElement>

function Paragraph({ children, as = 'p', ...otherProps }: Props) {
  return (
    <ParagraphStyle as={as} {...otherProps}>
      {children}
    </ParagraphStyle>
  )
}

export default Paragraph
