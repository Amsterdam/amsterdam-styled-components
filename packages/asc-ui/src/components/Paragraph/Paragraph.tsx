import React, { FunctionComponent } from 'react'
import ParagraphStyle, { Props } from './ParagraphStyle'

export { Props }

const Paragraph: FunctionComponent<
  Props & React.HTMLAttributes<HTMLParagraphElement>
> = ({ children, as, ...otherProps }) => (
  <ParagraphStyle forwardedAs={as} {...otherProps}>
    {children}
  </ParagraphStyle>
)

export default Paragraph
