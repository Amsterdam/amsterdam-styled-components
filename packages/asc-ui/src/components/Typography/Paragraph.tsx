import React from 'react'
import ParagraphStyle, { Props } from './ParagraphStyle'

export { Props }

const Paragraph: React.FC<Props> = ({ children }) => (
  <ParagraphStyle>{children}</ParagraphStyle>
)

export default Paragraph
