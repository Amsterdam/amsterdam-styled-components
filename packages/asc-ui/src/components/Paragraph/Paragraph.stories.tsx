import * as React from 'react'
import { storiesOf } from '@storybook/react'
import Paragraph from './Paragraph'

storiesOf('Atoms/Typography/Paragraph', module)
  .add('default style', () => (
    <>
      <Paragraph>
        Lorem ipsum dolor sit amet, consectetur adipisicing elit. Debitis
        deleniti dicta doloremque eveniet explicabo id ipsum omnis placeat rerum
        suscipit.
      </Paragraph>
    </>
  ))
  .add('strong style / intro text', () => (
    <>
      <Paragraph strong>
        Lorem ipsum dolor sit amet, consectetur adipisicing elit. Debitis
        deleniti dicta doloremque eveniet explicabo id ipsum omnis placeat rerum
        suscipit.
      </Paragraph>
    </>
  ))
  .add('long text (for editorial posts)', () => (
    <>
      <Paragraph hasLongText>
        Lorem ipsum dolor sit amet, consectetur adipisicing elit. Debitis
        deleniti dicta doloremque eveniet explicabo id ipsum omnis placeat rerum
        suscipit.
      </Paragraph>
    </>
  ))
