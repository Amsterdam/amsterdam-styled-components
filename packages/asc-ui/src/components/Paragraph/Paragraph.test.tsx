import { render } from '@testing-library/react'
import Paragraph from './Paragraph'

describe('Paragraph', () => {
  it('should render', () => {
    const { container } = render(<Paragraph>Foo</Paragraph>)

    expect(container.firstChild).toBeDefined()
  })
})
