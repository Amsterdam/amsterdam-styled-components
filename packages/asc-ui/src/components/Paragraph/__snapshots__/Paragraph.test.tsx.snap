// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Paragraph should render 1`] = `
.c0 {
  margin: 0;
  font-weight: 400;
  font-size: 16px;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 22px;
  margin-bottom: 20px;
  font-stretch: normal;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  margin-top: 0;
  color: inherit;
}

@media screen and (min-width:540.02px) {
  .c0 {
    margin-bottom: 24px;
  }
}

<p
  class="c0"
>
  Foo
</p>
`;
